src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -686,7 +703,9 @@ private async ValueTask<int> ReadAsyncInternal<TIOAdapter>(TIOAdapter adapter, M
 
                         if (status.ErrorCode == SecurityStatusPalErrorCode.Renegotiate)
                         {
-                            if (!_sslAuthenticationOptions!.AllowRenegotiation)
+                            // Tls1.3 does not have renegotiation. However on Windows this error code is used
+                            // for session management e.g. anything lsass needs to see.
+                            if (!_sslAuthenticationOptions!.AllowRenegotiation && SslProtocol != SslProtocols.Tls13)
```

Comment:
Is it possible that SslProtocol == SslProtocols.Tls13 and _handshakeWaiter != null?  Can we add asserts in various places?  It seems like things could break badly / hang if that invariant is broken.

Reply:
I don't think I fully understand your comment @stephentoub. The _handshakeWait will be always set in block above if status is SecurityStatusPalErrorCode.Renegotiate. That is regardless of protocol version. We either going to proceed with `ReplyOnReAuthenticationAsync()` or we fail if configuration disallows. Intention of this change is to always go with former option for TLS13. 

I _could_ move this check inside of the `lock (_handshakeLock)`.  It kind of feels silly to create waiter just to destroy it later. In the original PR I tries to limit what is done under the lock. But maybe this would be ok. Please let me know.

Reply:
Basically we're skipping completing the task here. I'm just wondering if that could leave things in a bad state in some cases, or if not, if there are asserts we should add to minimize the chances of a regression related to it.

Reply:
I was thinking more about it and I decided to rework the code little bit more. Now there is only one place where we SetResult() and that will hopefully make it easier to follow. Comment above was updated accordingly.